components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/messaging_reference.adoc

```diff
@@ -0,0 +1,451 @@
+[id="messaging_function_reference"]
+= Messaging function and event reference
+:description: Comprehensive function and event reference for the `StarknetMessaging` smart contracts.
+
+The `StarknetMessaging` smart contracts include functions and events that you use to implement various messaging flows in a dApp.
+
+For information on the messaging mechanism, see xref:architecture_and_concepts:Network_Architecture/messaging-mechanism.adoc[Messaging mechanism].
+
+The L1 functions, where available, are defined in the following smart contract:
+
+[cols=",,",]
+|===
+|Contract |Description |Functions
+
+|https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol[`StarknetMessaging.sol`] |
+The contract that implements messaging functions.
+
+Interfaces are available through https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/IStarknetMessaging.sol[`IStarknetMessaging.sol`]. a|
+* xref:#cancelL1ToL2Message[`cancelL1ToL2Message`]
+* xref:#consumeMessageFromL2[`consumeMessageFromL2`]
+* xref:#getMaxL1MsgFee[`getMaxL1MsgFee`]
+* xref:#l1ToL2Messages[`l1ToL2Messages`]
+* xref:#sendMessageToL2[`sendMessageToL2`]
+* xref:#startL1ToL2MessageCancellation[`startL1ToL2MessageCancellation`]
```

Comment:
these are only the functions defined in `IStarknetMessaging.sol` it misses the functions implemented in `StarknetMessaging.sol` which are not part of the interface.